kafka_consumer/tests/conftest.py

```diff
@@ -34,12 +34,55 @@ def initialize_topics():
 
 
 @pytest.fixture(scope='session')
-def dd_environment(e2e_instance):
+def mock_dns():
+    import socket
+
+    _orig_getaddrinfo = socket.getaddrinfo
+    _orig_connect = socket.socket.connect
+
+    def patched_getaddrinfo(host, *args, **kwargs):
+        if host == 'kafka1':
+            # See socket.getaddrinfo, just updating the hostname here.
+            # https://docs.python.org/3/library/socket.html#socket.getaddrinfo
+            return [(2, 1, 6, '', ('127.0.0.1', 9092))]
+        elif host == 'kafka2':
+            return [(2, 1, 6, '', ('127.0.0.1', 9093))]
+
+        return _orig_getaddrinfo(host, *args, **kwargs)
+
+    def patched_connect(self, address):
+        host, port = address[0], address[1]
+        if host in ('kafka1', 'kafka2'):
+            host = 'localhost'
+
+        return _orig_connect(self, (host, port))
+
+    socket.getaddrinfo = patched_getaddrinfo
+    socket.socket.connect = patched_connect
+    yield
+    socket.getaddrinfo = _orig_getaddrinfo
+    socket.socket.connect = _orig_connect
```

Comment:
```suggestion
    try:
        yield
    finally:
        socket.getaddrinfo = _orig_getaddrinfo
        socket.socket.connect = _orig_connect
```

Reply:
Not sure about this, if an exception is raised in the yield we should stop execution completely. That fixture should not catch errors raised during tests.

Reply:
It's better practice to be cautious but np